### DISCUSSION TITLE
Using rangelist() variables in constraints

### DISCUSSION THREAD

**Discussion by msmftc** [fvutils/pyvsc#126] [Q&A]
@mballance,

I have created randobj's that store rangelist values in their properties.  These properties are then used in constraints.  My problem is that updating the value of a rangelist-holding property to a new rangelist has no effect.  Constraints seem to remember only the initial rangelist that is stored in a property.

The example below illustrates the problem.  The Selector.available property is initialized to `vsc.rangelist((0, 999))`, and is used in a constraint.  Randomize() produces a stream of 10 integers in the range [0..999].  Then Selector.available is changed to `vsc.rangelist((1000, 1999))` and randomize() is called again, but the next 10 integers are still in the range [0..999], not [1000..1999].

Is this expected?  Is there a good workaround?  I realize that instead of being a rangelist, Selector.available could be of type list_t.  However, this would lose the efficiency representing large sets of contiguous integers in a rangelist.

```python
import vsc

@vsc.randobj
class Selector:
    def __init__(self):
        self.available = vsc.rangelist((0, 999))
        self.selected = vsc.rand_uint64_t()

    @vsc.constraint
    def max_c(self):
        self.selected.inside(self.available)

selector = Selector()
outStr = "A:"
for i in range(10):
    selector.randomize()
    outStr += "\t" + str(selector.selected)
print(outStr)

# Change the range that numbers are selected from.
selector.available = vsc.rangelist((1000, 1999))

outStr = "B:"
for i in range(10):
    selector.randomize()
    outStr += "\t" + str(selector.selected)
print(outStr)
```
```
A:      459     280     90      114     442     833     335     965     762     287
B:      413     966     921     486     733     688     285     999     864     886
```

**Comment by mballance** [comment]
@msmftc, interesting usage of rangelist! Certainly not one I had in mind when implementing the construct, but seems useful and valid to me. The core issue, here, is that rangelist doesn't currently provide a mechanism for updating the rangelist data model. Assigning to the attribute simply points 'available' to a new rangelist while the constraint is still looking at the old one.
I'll create an enhancement request for this and plan to look at it in the near future. Shouldn't be too difficult to support updating a rangelist...
